docs/index.rst

```diff
@@ -2,58 +2,85 @@
+- the base container image
+- the version of Python
+- the version of ansible-core
+- the version of ansible-runner
+- Ansible collections, with version restrictions
+- system packages, with version restrictions
+- Python packages, with version restrictions
+- other items to download, install, or configure
+
+Choosing a base image
+---------------------
+
+You can use any base image you choose. The smaller the base image, generally, the smaller the final image. However, to make Ansible Builder more efficient, you should know what packages, if any, are already installed on the base image you use. For example, some base images already have Python installed. Others do not. If you use a base image that already has Python installed, you can omit Python in your execution environment definition file. Not all base images have package managers installed.
 
-.. _ansible-runner: https://github.com/ansible/ansible-runner
+How Ansible Builder executes
+============================
 
+Ansible Builder can execute two separate steps. The first step is to create a build instruction file (Containerfile for Podman, Dockerfile for Docker) and a build context based on the execution environment definition file. The second step is to run a containerization tool (Podman or Docker) to build an image based on the build instruction file and build context. The ``ansible-builder build`` command runs both steps. The ``ansible-builder create`` command runs only the first step. For more details, read through the :ref:`CLI usage docs <builder_cli>`.
 
-Getting Started
-***************
+How Ansible Builder builds images
+---------------------------------
 
-To get started with Ansible Builder, start by creating an
-:ref:`Execution Environment definition<Definition:Execution Environment Definition>`. This
-file allows you to specify which content you would like to include in your
-execution environment, such as collections, Python requirements, and
-system-level packages.
+Ansible Builder executes four stages when it runs your containerization tool to build a container image. The same four stages get executed if you build your container image directly with Podman or Docker, using a build instruction file and context generated by ``ansible-builder create``. These stages are:
+- Base
+- Galaxy
+- Builder
+- Final
+In the Base stage, Ansible Builder uses Podman or Docker to pull the base image you defined. All three later stages of the build process build on the output of the Base stage. In the Galaxy stage, Ansible Builder downloads the collections you defined from Galaxy and stashes them locally as files. In the Builder stage, Ansible Builder downloads the other packages (Python packages and system packages) you defined and stashes them locally as files. In the Final stage, Ansible Builder integrates the first three stages, installing all the stashed files on the base image and generating a new image that includes all the content.
```

Comment:
During the base stage, not only is the base image pulled, but it is also customized for all later stages. E.g., ansible and ansible-runner are installed (if defined), alternate Python version is installed (if defined), and we make sure pip is available. There are no "stashed files" on the base image, either. Those are only from Galaxy and Builder intermediate stages.

Reply:
Ah, thanks for the details! I hope 579613fefc2e2fe8da699244ed05e22552b22742 addresses all these comments @Shrews.